flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedApproximateSubpartitionView.java

```diff
@@ -0,0 +1,27 @@
+package org.apache.flink.runtime.io.network.partition;
+
+import static org.apache.flink.util.Preconditions.checkState;
+
+public class PipelinedApproximateSubpartitionView extends PipelinedSubpartitionView {
+
+	PipelinedApproximateSubpartitionView(PipelinedApproximateSubpartition parent, BufferAvailabilityListener listener) {
+		super(parent, listener);
+	}
+
+	@Override
+	public void releaseAllResources() {
+		if (isReleased.compareAndSet(false, true)) {
+			// The view doesn't hold any resources and the parent cannot be restarted. Therefore,
+			// it's OK to notify about consumption as well.
+			checkState(parent instanceof PipelinedApproximateSubpartition);
+			((PipelinedApproximateSubpartition) parent).releaseView();
+		}
+	}
```

Comment:
Why is this method behaving differently compared to `PipelinedSubpartitionView#releaseAllResources`?

Reply:
PipelinedSubpartitionView#releaseAllResources
```
        @Override
	public void releaseAllResources() {
		if (isReleased.compareAndSet(false, true)) {
			// The view doesn't hold any resources and the parent cannot be restarted. Therefore,
			// it's OK to notify about consumption as well.
			parent.onConsumedSubpartition();
		}
	}
```

They are different in the sense that `PipelinedSubpartitionView` release the subPartition => Partition as well
`PipelinedApproximateSubpartition` only releases view.